datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -514,10 +521,13 @@ pub fn statistics_from_parquet_meta_calc(
     statistics.total_byte_size = Precision::Exact(total_byte_size);
 
     let file_metadata = metadata.file_metadata();
-    let file_schema = parquet_to_arrow_schema(
+    let mut file_schema = parquet_to_arrow_schema(
```

Comment:
I messed around with this and I agree this is the least invasive change for supporting string statistics

I feel like there must be a better way, however. I will think about it

Reply:
> I feel like there must be a better way, however.

I agree. I don't like passing the config around either...

Reply:
Follow up ticket filed https://github.com/apache/datafusion/issues/12123